presto-native-execution/presto_cpp/main/http/tests/HttpTestBase.h

```diff
@@ -185,6 +185,7 @@ class HttpClientFactory {
   }
 
   ~HttpClientFactory() {
+    eventBase_->runInEventBaseThread([pools = std::move(sessionPools_)] {});
```

Comment:
Because it is pinned to one event base and the destructor need to release some resource associated with it.  It is documented in the comment of the class.